mesa/space.py

```diff
@@ -311,10 +311,7 @@ def iter_cell_list_contents(
             An iterator of the contents of the cells identified in cell_list
 
         """
-        for (x, y) in cell_list:
-            content = self.grid[x][y]
-            if not content == self.default_val():
-                yield content
+        return filter(None, (self.grid[x][y] for x, y in cell_list))
```

Comment:
`default_val()` is not always `None`, why the hardcode to `None`?

Reply:
This is not a hard-code to None. The filter function in Python is a bit unusual in the way that instead of a filter function you can use None to filter anything that bools to True (equivalent to `item for item in iterable if item`)

In this context `None, set() and list()` are all False.

Check the docs at [https://docs.python.org/3/library/functions.html#filter]()

Reply:
I don't think `filter` will offer any speedup over list comprehension. Last time I tested, list comprehension was a bit faster even.

Reply:
This is not about speedup from filter vs list comprehension. The equivalent generator expression would be:
```python
(self[x][y] for x, y in cell_list if self[x][y])
```
However this would call/access `self[x][y]` twice, which makes this slower.